src/operator/nn/softmax-inl.h

```diff
@@ -310,6 +313,54 @@ void SoftmaxCompute(const nnvm::NodeAttrs& attrs,
   });
 }
 
+#if MSHADOW_USE_MKL == 1
+static inline int64_t prod(int64_t* n, int start, int end) {
+  int64_t res = 1;
+  for (int i = start; i < end; i++)
+    res *= n[i];
+  return res;
+}
+
+static inline void max_(int64_t n, float * __restrict__ in, float *dst) {
```

Comment:
cblas_isamax() might have better perf.

Reply:
Seems cblas_isamax() is for absolute values. Not sure how to use here. Any suggestion?

https://software.intel.com/en-us/mkl-developer-reference-c-cblas-i-amax

Reply:
You're right, looks like no available max/min function for both signed and unsigned values from MKL. :)